File: Mu2e-STMDAQ/OLD_BUT_DONT_DELETE_YET/NOT_NEEDED/hardwareScripts/FMC144/capture.cc
////////////////////////////////////////////////////////////
/// Capture module: interface with axi_capture star (main)
////////////////////////////////////////////////////////////

// Capture
#include "STMDAQ-TestBeam/hardwareScripts/FMC144/capture.hh"

//Standard constructor - shouldn't be used
capture::capture() {}

// Set the burst size in axi_capture size
int capture::setburstlength(IPBusManager* hw, int capturenode_nr, int numberburst, int burstlength){

  std::string node;

  node ="Captures.capture_"+std::to_string(capturenode_nr)+".nb_of_bursts";
  hw->write(node,numberburst);

  node ="Captures.capture_"+std::to_string(capturenode_nr)+".burst_length";
  hw->write(node,burstlength);

  return 1;

}

// Enable uploading  
int capture::enable_upload(IPBusManager* hw, int capturenode_nr){

  std::string node ="Captures.capture_"+std::to_string(capturenode_nr)+".ctrl";
  hw->write(node,0x01);

  return 1;

}

// Enable uploading with hardware trigger  
int capture::enable_upload_hw_trig(IPBusManager* hw, int capturenode_nr){

  std::string node ="Captures.capture_"+std::to_string(capturenode_nr)+".ctrl";
  hw->write(node,0x03);

  return 1;

}

// Disable uploading 
int capture::disable_upload(IPBusManager* hw, int capturenode_nr){

  std::string node ="Captures.capture_"+std::to_string(capturenode_nr)+".ctrl";
  hw->write(node,0x00);

  return 1;

}

// Arm the Unit  
int capture::arm(IPBusManager* hw, int capturenode_nr){

  std::string node ="Captures.capture_"+std::to_string(capturenode_nr)+".command";
  hw->write(node,0x01);

  return 1;

}

// Get the rx and tx pointers from the FIFOs  
uint64_t capture::fifo_pointer_val(IPBusManager* hw, int capturenode_nr){

  std::string node ="Captures.capture_"+std::to_string(capturenode_nr)+".status";
  uint64_t value = hw->read(node);

  return value;

}

// Send a software (sw) trigger 
int capture::sw_trigger(IPBusManager* hw, int capturenode_nr){

  std::string node ="Captures.capture_"+std::to_string(capturenode_nr)+".command";
  hw->write(node,0x08);

  return 1;

}

// Enable external trigger in the unit 
int capture::external_trigger_enable(IPBusManager* hw, int capturenode_nr){

  std::string node ="Captures.capture_"+std::to_string(capturenode_nr)+".ctrl";
  hw->write(node,0x02);

  return 1;

}

// Reset FIFOs  
int capture::fifo_rst(IPBusManager* hw, int capturenode_nr){

  std::string node ="Captures.capture_"+std::to_string(capturenode_nr)+".command";
  hw->write(node,0x10);

  return 1;

}

File: Mu2e-STMDAQ/OLD_BUT_DONT_DELETE_YET/NOT_NEEDED/hardwareScripts/FMC144/pythonSource/capture.py
#!/usr/bin/python

# -*- coding: utf-8 -*-
import sys
import time
import uhal
import spi_functions as s

def capture_setburstlength (hw,capturenode_nr,numberburst,burstlength):
  
  hw.getNode("Captures.capture_%s.nb_of_bursts" %(capturenode_nr)).write(numberburst)
  hw.dispatch()
  hw.getNode("Captures.capture_%s.burst_length" %(capturenode_nr)).write(burstlength)
  hw.dispatch()
  
  return 1

def capture_enable_upload (hw,capturenode_nr):
  hw.getNode("Captures.capture_%s.ctrl" %(capturenode_nr)).write(0x01)
  hw.dispatch()
  
  return 1

def capture_enable_upload_hw_trig (hw,capturenode_nr):
  hw.getNode("Captures.capture_%s.ctrl" %(capturenode_nr)).write(0x03)
  hw.dispatch()
  
  return 1


def capture_disable_upload (hw,capturenode_nr):
  hw.getNode("Captures.capture_%s.ctrl" %(capturenode_nr)).write(0x00)
  hw.dispatch()
  
  return 1
  
def capture_arm(hw,capturenode_nr):
  hw.getNode("Captures.capture_%s.command" %(capturenode_nr)).write(0x01)
  hw.dispatch()
  
  return 1
  
def capture_fifo_pointer_val(hw,capturenode_nr):
  value = hw.getNode("Captures.capture_%s.status" %(capturenode_nr)).read()
  hw.dispatch()
  
  return value

def capture_sw_trigger(hw,capturenode_nr):
  hw.getNode("Captures.capture_%s.command" %(capturenode_nr)).write(0x08)
  hw.dispatch()
  
  return 1

def capture_external_trigger_enable(hw,capturenode_nr):
  hw.getNode("Captures.capture_%s.ctrl" %(capturenode_nr)).write(0x02)
  hw.dispatch()
  
  return 1

def capture_fifo_rst(hw,capturenode_nr):
  hw.getNode("Captures.capture_%s.command" %(capturenode_nr)).write(0x10)
  hw.dispatch()
  
  return 1
